modules/engine/src/index.ts

```diff
@@ -1199,15 +1115,15 @@ export class VectorEngine implements IVectorEngine {
       "both",
     );
     if (sigRes.isError) {
-      return sendResponseToCounterparty(RestoreError.reasons.InvalidSignatures, {
+      return generateRestoreError(RestoreError.reasons.InvalidSignatures, {
         recoveryError: sigRes.getError().message,
       });
     }
 
     // Verify transfers match merkleRoot
     const { root } = generateMerkleTreeData(activeTransfers);
     if (root !== channel.merkleRoot) {
-      return sendResponseToCounterparty(RestoreError.reasons.InvalidMerkleRoot, {
+      return generateRestoreError(RestoreError.reasons.InvalidMerkleRoot, {
```

Comment:
Everywhere `generateMerkleTreeData` is called needs to call `.free()` on the tree at some point or memory leaks. See above: `const { root } = gen...`. This needs to be `const { root, tree } = gen...; tree.free();`

There are a few cases of this, so please review all instances of `generateMerkleTreeData`

Reply:
Hmm... actually I think this is fixed in #585 . I'm starting to get confused by the multiple PRs in flight.

Reply:
Yeah, this PR should be considered the "master PR" that will get merged when we are ready to cut a release. All other PRs are proposed to be merged into this branch specifically. However, this would likely not get fixed by #585 since this is in the engine, and that is for the protocol, so I'll add it in now.